w3-aggregation.md

```diff
@@ -0,0 +1,337 @@
+
+### Storefront offers broker an aggregate
+
+When a Storefront has enough content to fulfill an aggregate (each broker MAY have different requirements), a Filecoin deal for an aggregate SHALL be requested by a `aggregate/offer` invocations. Deal negotiations with Filecoin Storage Providers will be handled out off band. Broker SHOULD generate a receipt to acknowledge the received request.
+
+```mermaid
+sequenceDiagram
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Storefront->>Authority: invoke `aggregate/offer`
+    Note left of Authority: Request offer to be queued
+```
+
+### Broker queues the offer
+
+Once broker successfully gets an offer, the offer gets queued for review. A receipt is created to proof the transition of `aggregate/offer` state from `null` into `queued`.
+
+This receipt MUST contain a follow up task in the (`.fx.join` field) that is run when submitted request is processed. It MAY succeed (if aggregate was accepted) or fail (if aggregated was  determined to be invalid). The result of the subsequent task CAN be looked up using its receipt.
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+
+    Storefront->>Authority: invoke `aggregate/offer`
+    Note left of Storefront: Queueing offer
+    Authority-->>Storefront: receipt issued
+```
+
+### Broker reviews and handles the offer
+
+When a broker pops the offer from the queue, the offer details MUST be retrievable. With the offer details, the broker MAY interact with available Filecoin Storage Providers, in order to establish a previously determined number of deals. Depending on storage providers availability, as well as the content present in the offer, the aggregate MAY be handled or not. A receipt is created to proof the transition of `aggregate/offer` state from `queued` into `accepted` or `denied`.
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+
+    Note left of Storefront: Review and handle offer async
+    Authority-->>Storefront: receipt issued
+```
+
+Once offer is accepted broker takes care of arranging and renewing deals.
+
+### Storefront can query state of the aggregate deals
+
+Storefront users MAY want to check details about deals from the content they previously stored. These deals will change over time as they get renewed. Therefore, Storefront should invoke `aggregate/get` capability to gather information about given aggregate identifier.
```

Comment:
For me, `aggregate/get` suggests we're retrieving the aggregate data, which is not the case, we're retrieving information _about_ the deals the aggregate is included in.

`aggregate/info` `aggregate/status` or `aggregate/deals` (I'm guessing this one was disapproved already)

Reply:
everything is a bit overloaded terms. It depends on what we consider an aggregate. If we do not assemble one (is just metadata about one in the new world, given Spade also not assemble it...), I would not expect get to get the assembled thing.

We can align on naming when we can sync and update if needed

Reply:
I agree! Lets separate general review from the naming which we can do in a separately in sync.